src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -508,25 +462,50 @@ private bool CheckLocalValueKind(SyntaxNode node, BoundLocal local, BindValueKin
             return true;
         }
 
-        private bool CheckParameterValueKind(SyntaxNode node, BoundParameter parameter, BindValueKind valueKind, bool checkingReceiver, DiagnosticBag diagnostics)
+        private static bool CheckLocalRefEscape(SyntaxNode node, BoundLocal local, uint escapeTo, bool checkingReceiver, DiagnosticBag diagnostics)
         {
-            ParameterSymbol parameterSymbol = parameter.ParameterSymbol;
-            var paramKind = parameterSymbol.RefKind;
+            LocalSymbol localSymbol = local.LocalSymbol;
+
+            if (localSymbol.RefEscapeScope <= escapeTo)
```

Comment:
Can you refactor this comparison into a method? I don't know what it means for an escape scope to be `<=` to another one.

Reply:
It seems this comparison is used in only one place. I will add a comment instead:
            // if local symbol can escape to the same or wider/shallower scope then escapeTo
            // then it is all ok, otherwise it is an error.


---
In reply to: [135154511](https://github.com/dotnet/roslyn/pull/21334#discussion_r135154511) [](ancestors = 135154511)